### PR TITLE
feat: workflow to update and build

### PR DESCRIPTION
**Pull request by oliver-butterley**
It seemed that it could be useful to have a custom workflow that can run everyday, check for updates and deploy the updated version directly. Here is a first attempt at such. 

I have tested it a bit and seems fine although I haven't tested very extensively. Let me know if something like this is useful and if you have any requests for modifications, etc.

### CODE REVIEW

**Comment by oliver-butterley** [comment]
I realise now that two changes that probably should be made:
- Change the git user to the github-actions[bot]
- Run only on the main branch (unfortunately the action has already been running on the PR branch and changing things)

File: .github/workflows/update.yaml [hunk 1]
```diff
@@ -1,26 +1,82 @@
...
+    - cron: "0 8 * * *" # every day at 8:00 UTC
+  workflow_dispatch: # allows workflow to be triggered manually
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
 
 jobs:
-  update_lean:
+  update:
```
Comment (Seasawher) [inline .github/workflows/update.yaml:14]: Run on main branch only.

```suggestion
  update:
    if: github.ref == 'refs/heads/main'
```

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -6,10 +6,6 @@ concurrency:
   cancel-in-progress: true
 
 on:
```
Comment (Seasawher) [inline .github/workflows/build.yaml:8]: Changes to build.yaml should be reverted. When changes are made to the python script and pushed, a GitHub Action must be run automatically.

File: .github/workflows/update.yaml [hunk 3]
```diff
@@ -1,26 +1,82 @@
...
+        uses: actions/checkout@v4
+
+      - name: Setup mdbook
+        uses: peaceiris/actions-mdbook@v2
         with:
-          #  Allowed values: "silent", "commit", "issue" or "pr". Default: "commit".
-          on_update_succeeds: pr
-          # Allowed values: "silent", "issue" or "fail". Default: "issue".
-          on_update_fails: issue
+          mdbook-version: "latest"
```
Comment (Seasawher) [inline .github/workflows/update.yaml:70]: use the same version to `build.yaml`
```suggestion
          mdbook-version: '0.4.32'
```

File: .github/workflows/update.yaml [hunk 4]
```diff
@@ -1,26 +1,82 @@
...
+            echo "files_changed=true"  >> "$GITHUB_OUTPUT"
+           else
+            echo "files_changed=false" >> "$GITHUB_OUTPUT"
+          fi
+          cat "$GITHUB_OUTPUT"
+
+      - name: Commit updated version
+        if: steps.update-lean-mathlib.outputs.result == 'update-success'
+        run: |
+          git config user.name "$GITHUB_ACTOR"
```
Comment (Seasawher) [inline .github/workflows/update.yaml:44]: I want to use `action@github.com` for commiter. (same as `build.yaml`)

**Comment by oliver-butterley** [comment]
Concerning the git user, after [some reading](https://github.com/actions/checkout/pull/1707) it appears the best option is:
```bash
git config user.name "github-actions[bot]"
git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
```
However any option still does the job in this particular use case.

On the other hand, the commit step could be completely removed. It could be arranged in a way that the only thing done is that every day an updated version is deployed to pages. In this sense an updated version of the repo is not required. 

Let me know your preference and I can update the workflow correspondingly. My only objective was that the list of tactics on the page is always up to date with zero effort from humans.

**Comment by Seasawher** [comment]
> On the other hand, the commit step could be completely removed. It could be arranged in a way that the only thing done is that every day an updated version is deployed to pages. In this sense an updated version of the repo is not required.

The commit step is necessary. The reason is that a markdown file is in this repository. I check the differences by looking at the markdown file.

**Comment by Seasawher** [comment]
> Concerning the git user, after [some reading](https://github.com/actions/checkout/pull/1707) it appears the best option is:

Thank you! However, for uniformity, please use the same commit method as in `build.yml`.

**Pull request merged**